Build issue with LLVM 18.1.1 on msys2

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Affected Build(s)

HEAD

### Description of Issue

Building Citra with Clang 18.1.1 on msys2 now yields this error:

```
In file included from C:/msys64/home/Default/github/bravely-offline-citra/src/citra_qt/util/graphics_device_info.cpp:5:
In file included from C:/msys64/home/Default/github/bravely-offline-citra/src/./citra_qt/util/graphics_device_info.h:8:
In file included from C:/msys64/clang64/include/qt6/QtCore/QString:1:
In file included from C:/msys64/clang64/include/qt6/QtCore/qstring.h:14:
In file included from C:/msys64/clang64/include/qt6/QtCore/qchar.h:656:
In file included from C:/msys64/clang64/include/qt6/QtCore/qstringview.h:8:
In file included from C:/msys64/clang64/include/qt6/QtCore/qbytearray.h:14:
C:/msys64/clang64/include/qt6/QtCore/qbytearrayview.h:69:27: error: 'char_traits<unsigned char>' is deprecated: char_traits<T> for T not equal to char, wchar_t, char8_t, char16_t or char32_t i
s non-standard and is provided for a temporary period. It will be removed in LLVM 19, so please migrate off of it. [-Werror,-Wdeprecated-declarations]
   69 |     return qsizetype(std::char_traits<Char>::length(data));
      |                           ^
C:/msys64/clang64/include/qt6/QtCore/qbytearrayview.h:151:39: note: in instantiation of function template specialization 'QtPrivate::lengthHelperPointer<unsigned char>' requested here
  151 |               data, data ? QtPrivate::lengthHelperPointer(data) : 0) {}
      |                                       ^
C:/msys64/home/Default/github/bravely-offline-citra/src/citra_qt/util/graphics_device_info.cpp:25:34: note: in instantiation of function template specialization 'QByteArrayView::QByteArrayView
<const unsigned char *, true>' requested here
   25 |         return QString::fromUtf8(context.functions()->glGetString(GL_RENDERER));
      |                                  ^
C:/msys64/clang64/include/c++/v1/__string/char_traits.h:81:8: note: 'char_traits<unsigned char>' has been explicitly marked deprecated here
   81 | struct _LIBCPP_DEPRECATED_(
      |        ^
C:/msys64/clang64/include/c++/v1/__config:975:53: note: expanded from macro '_LIBCPP_DEPRECATED_'
  975 | #      define _LIBCPP_DEPRECATED_(m) __attribute__((__deprecated__(m)))
      |                                                     ^
1 error generated.
ninja: build stopped: subcommand failed.
```
And the compiler seems to now be enforcing this deprecation.

I'm not a strong coder, but I was able to get something that compiled with the assistance of Copilot AI, but it breaks Vulkan (so more investigation needed, but I get better performance on OpenGL on this set up anyway so I can live with it).

That said, as far as I can tell, _at least_ these files need to be modified:

```
src/citra_qt/main.cpp
src/citra_qt/util/graphics_device_info.cpp
src/citra_qt/util/graphics_device_info.h
```

to rework the QString stuff. I'm not very familiar with the code to know if there are other places as well.

### Expected Behavior

Successful compilation.

### Reproduction Steps

Build Citra with Clang 18.1.1 (on msys2, but maybe on other OSes as well).

### Log File

No logs; this is a compile issue.

### System Configuration

CPU: Intel Kaby Lake
GPU/Driver: Integrated Intel HD Graphics 620
RAM: 8GB
OS: Windows 10 64-bit

--- a/src/audio_core/openal_sink.cpp
+++ b/src/audio_core/openal_sink.cpp
@@ -73,8 +73,9 @@ OpenALSink::OpenALSink(std::string device_name) : impl(std::make_unique<Impl>())
 
     auto alBufferCallbackSOFT =
         reinterpret_cast<LPALBUFFERCALLBACKSOFT>(alGetProcAddress("alBufferCallbackSOFT"));
-    alBufferCallbackSOFT(impl->buffer, AL_FORMAT_STEREO16, native_sample_rate, &Impl::Callback,
-                         impl.get());
+    alBufferCallbackSOFT(impl->buffer, AL_FORMAT_STEREO16, native_sample_rate,
+                         reinterpret_cast<ALBUFFERCALLBACKTYPESOFT>(&Impl::Callback), impl.get());
+
     if (alGetError() != AL_NO_ERROR) {
         LOG_CRITICAL(Audio_Sink, "alBufferCallbackSOFT failed: {}", alGetError());
         Close();

--- a/src/citra_qt/util/graphics_device_info.cpp
+++ b/src/citra_qt/util/graphics_device_info.cpp
@@ -22,7 +22,7 @@ QString GetOpenGLRenderer() {
     QOpenGLContext context;
     if (context.create()) {
         context.makeCurrent(&surface);
-        return QString::fromUtf8(context.functions()->glGetString(GL_RENDERER));
+        return QString::fromUtf8(reinterpret_cast<const char*>(context.functions()->glGetString(GL_RENDERER)));
     } else {
         return QStringLiteral("");
     }

--- a/src/citra_qt/util/graphics_device_info.cpp
+++ b/src/citra_qt/util/graphics_device_info.cpp
@@ -22,7 +22,8 @@ QString GetOpenGLRenderer() {
     QOpenGLContext context;
     if (context.create()) {
         context.makeCurrent(&surface);
-        return QString::fromUtf8(reinterpret_cast<const char*>(context.functions()->glGetString(GL_RENDERER)));
+        return QString::fromUtf8(
+            reinterpret_cast<const char*>(context.functions()->glGetString(GL_RENDERER)));
     } else {
         return QStringLiteral("");
     }

--- a/src/core/hle/service/soc/soc_u.cpp
+++ b/src/core/hle/service/soc/soc_u.cpp
@@ -2251,7 +2251,7 @@ std::optional<SOC_U::InterfaceInfo> SOC_U::GetDefaultInterfaceInfo() {
     socklen_t s_info_len = sizeof(struct sockaddr_in);
     sockaddr_in s_info;
 
-    if ((sock_fd = ::socket(AF_INET, SOCK_STREAM, 0)) == -1) {
+	if (static_cast<int>(sock_fd = ::socket(AF_INET, SOCK_STREAM, 0)) == -1) {
         return std::nullopt;
     }
 
@@ -2269,7 +2269,7 @@ std::optional<SOC_U::InterfaceInfo> SOC_U::GetDefaultInterfaceInfo() {
 
 #ifdef _WIN32
     sock_fd = WSASocket(AF_INET, SOCK_DGRAM, 0, 0, 0, 0);
-    if (sock_fd == SOCKET_ERROR) {
+    if (static_cast<int>(sock_fd) == SOCKET_ERROR) {
         return std::nullopt;
     }
 

--- a/src/core/hle/service/soc/soc_u.cpp
+++ b/src/core/hle/service/soc/soc_u.cpp
@@ -2251,7 +2251,7 @@ std::optional<SOC_U::InterfaceInfo> SOC_U::GetDefaultInterfaceInfo() {
     socklen_t s_info_len = sizeof(struct sockaddr_in);
     sockaddr_in s_info;
 
-	if (static_cast<int>(sock_fd = ::socket(AF_INET, SOCK_STREAM, 0)) == -1) {
+    if (static_cast<int>(sock_fd = ::socket(AF_INET, SOCK_STREAM, 0)) == -1) {
         return std::nullopt;
     }
 

